### PR TITLE
Code changes to support ppc64le platform

### PR DESCRIPTION
**Pull request by sunil1783**
Signed-off-by: Sunil1783 <sunil.botre@ibm.com>

What type of PR is this?
kind/feature

Any specific area of the project related to this PR?
area/build

What this PR does / why we need it:
These code changes will support falco on Ubuntu 18.04 ppc64le platform.

Which issue(s) this PR fixes:
None

Special notes for your reviewer:
   These code changes will support falco code on Ubuntu 18.04 ppc64le platform. 
   Enabled the -DBUILD_BPF=True while compiling the code using CMAKE to provide eBPF support.

   cmake  -DBUILD_BPF=True -DCMAKE_VERBOSE_MAKEFILE=On
   -DCPACK_SOURCE_RPM=ON -DUSE_BUNDLED_DEPS=ON  ..

Does this PR introduce a user-facing change?

```release-note
new: support for ppc64le platform
```

### CODE REVIEW

**Comment by fntlnz** [comment]
Hi @sunil1783 - I'd like falco to support the ppc64le platform. However the way you are doing it here completely drops support for x86. We can discuss on a way to design this so that both are kept and many more can come.

File: docker/tester/root/usr/bin/entrypoint [hunk 1]
```diff
@@ -27,10 +27,11 @@ case "$CMD" in
         echo "Missing Falco version." >&2
         exit 1
     fi
+
     PACKAGE="$BUILD_DIR/$BUILD_TYPE/falco-$FALCO_VERSION-x86_64.deb"
     if [ ! -f "$PACKAGE" ]; then
-        echo "Package(s) not found." >&2
-        exit 1
+	echo "Package(s) not found." >&2
```
Comment (leodido) [inline docker/tester/root/usr/bin/entrypoint]: Why tabs here?

**Comment by sunil1783** [comment]
Created issue for test case failures on ppc64le. 
Currently all the falco test cases are passed, except test case which required falcosecurity/falco:test docker image.
https://github.com/falcosecurity/falco/issues/982.

Need help/suggetions to port the Dockefiles for ppc64le platform which downloads Intel specific gcc6&5 dependent pre-built packages.

File: cmake/modules/OpenSSL.cmake [hunk 2]
```diff
@@ -14,6 +14,7 @@ else()
   set(OPENSSL_LIBRARY_SSL "${OPENSSL_INSTALL_DIR}/lib/libssl.a")
   set(OPENSSL_LIBRARY_CRYPTO "${OPENSSL_INSTALL_DIR}/lib/libcrypto.a")
   set(OPENSSL_BINARY "${OPENSSL_INSTALL_DIR}/bin/openssl")
+  set(USE_BUNDLED_OPENSSL ON)
```
Comment (leodido) [inline cmake/modules/OpenSSL.cmake]: Does this needs to be always ON?

We enable it only when USE_BUNDLED_DEPS is ON, for cmake/modules/sysdig.cmake dependency.

File: cmake/modules/sysdig.cmake [hunk 3]
```diff
@@ -22,7 +22,12 @@ endif()
 
 file(MAKE_DIRECTORY ${SYSDIG_CMAKE_WORKING_DIR})
 # cd /path/to/build && cmake /path/to/source
-execute_process(COMMAND "${CMAKE_COMMAND}" ${SYSDIG_CMAKE_SOURCE_DIR} WORKING_DIRECTORY ${SYSDIG_CMAKE_WORKING_DIR})
+
+if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
+    execute_process(COMMAND "${CMAKE_COMMAND}" ${SYSDIG_CMAKE_SOURCE_DIR} WORKING_DIRECTORY ${SYSDIG_CMAKE_WORKING_DIR})
+elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "ppc64le")
+    execute_process(COMMAND "${CMAKE_COMMAND}" "-DSYSDIG_VERSION=dev" ${SYSDIG_CMAKE_SOURCE_DIR} WORKING_DIRECTORY ${SYSDIG_CMAKE_WORKING_DIR})
```
Comment (leodido) [inline cmake/modules/sysdig.cmake]: Why enforcing `dev` version for sysdig dependency here?

File: test/falco_test.py [hunk 4]
```diff
@@ -426,8 +427,13 @@ def test(self):
         self.possibly_copy_driver()
 
         if self.package != 'None':
-            # This sets falco_binary_path as a side-effect.
-            self.install_package()
+            #As per https://github.com/falcosecurity/falco/issues/982, skipping the docker_package test cases
+            #due to falcobuilder/falco-tester container, old kernel versions module build issue(gcc-6/5 dependecy) on ppc64le platform
+            if platform.machine() == 'ppc64le' and self.package.startswith("docker:"):
```
Comment (leodido) [inline test/falco_test.py]: I suspect this check will disable the testing for both DEB and RPM ((ref)[https://github.com/falcosecurity/falco/blob/3693b16c917d6920adffac03e80d38c1b64317ea/test/falco_tests_package.yaml]).

Maybe I'm remembering it wrong, but this definitely needs to be double-checked :)

File: userspace/falco/falco.cpp [hunk 5]
```diff
@@ -295,16 +295,32 @@ uint64_t do_inspect(falco_engine *engine,
 			throw sinsp_exception(inspector->getlasterr().c_str());
 		}
 
-		if (duration_start == 0)
-		{
-			duration_start = ev->get_ts();
-		} else if(duration_to_tot_ns > 0)
-		{
-			if(ev->get_ts() - duration_start >= duration_to_tot_ns)
+                #if defined(__x86_64__)
```
Comment (leodido) [inline userspace/falco/falco.cpp]: This needs some attention, too.

Can we find a way to make it work on both the platforms without using `#if#elsif defined(...)` ?

**Comment by fntlnz** [comment]
Let's keep this open since it has good points for the tests too - we need to decide the direction with #1225

**Pull request closed without merging**